client/src/containers/header/notifications/logic/saga.ts

```diff
@@ -2,26 +2,125 @@ import { all, put, select, takeEvery } from 'redux-saga/effects';
 import { IRootState } from 'typings/root-state';
 import * as actions from './actions';
 import { NotificationsActionTypes } from './action-types';
-import { editNotification } from './actions';
-
+import { getFirestore, collection, query, getDocs, limit, orderBy } from 'firebase/firestore';
+import { app } from 'containers/app/app';
+import { NotificationTypes, TNotification } from 'typings/common/INotification';
+import { doc, setDoc, where } from 'firebase/firestore';
 function* readNotification({ id }: ReturnType<typeof actions.readNotification>) {
 	const store: IRootState = yield select();
+	const firestore = getFirestore(app);
 	const notification = store.header.notifications.get(id);
+	const userId = store.auth.userData.user?.id;
+
 	if (notification) {
-		// TODO: socket.io
+		yield setDoc(doc(firestore, 'notifications', id), {
+			createdAt: notification.date,
+			id: notification.id,
+			body: notification.body,
+			type: notification.type,
+			read: true,
+			userId,
+		});
 		yield put(
-			editNotification({
+			actions.editNotification({
 				id: notification.id,
 				value: { ...notification, read: true },
 			}),
 		);
 	}
 }
 
+function* fetchNotification() {
+	const firestore = getFirestore(app);
+	const store: IRootState = yield select();
+	const userId = store.auth.userData.user?.id;
+	const q = query(
+		collection(firestore, 'notifications'),
+		where('userId', '==', userId),
+		orderBy('createdAt'),
+		limit(10),
+	);
+	const querySnapshot: TNotification[] = [];
+	const result: Record<string, any>[] = yield getDocs(q);
+	store.header.notifications.clear();
+	result.forEach((notification) => {
+		const data = notification.data();
+		switch (data.type) {
+			case NotificationTypes.Common:
```

Comment:
A lot of same objects are used. I think, you should make one initial and then use `{...initial, body: {...}}`